Source/Core/Common/StringUtil.cpp

```diff
@@ -554,17 +562,9 @@ std::string UTF8ToSHIFTJIS(const std::string& input)
   return CodeTo("SJIS", "UTF-8", input);
 }
 
-std::string UTF16ToUTF8(const std::wstring& input)
+std::string UTF16BEToUTF8(const char16_t* str, size_t max_size)
 {
-  return CodeToUTF8("UTF-16LE", input);
+  return CodeToUTF8("UTF-16BE", std::u16string(str, max_size));
```

Comment:
Doesn't this `u16string` constructor always read `max_size` code units? The number of code units that this function is supposed to read is whichever is smallest out of `max_size` and the position of the first null code unit. So I think you still need the line `const char16_t* str_end = std::find(str, str + max_size, '\0');`.

Reply:
It should then should be possible to construct the string using `std::u16string(str, str_end)`, but please correct me if I'm wrong.

Reply:
Correct again! Sorry for the hand-holding.

Reply:
Don't worry about it. I'm just happy that someone cares enough about this issue to get it fixed :)